### DISCUSSION TITLE
getDefinitionsProductType API  Variation_Theme

### DISCUSSION THREAD

**Discussion by wszwc** [amzn/selling-partner-api-models#3814] [General]
Hello guys, I have a very serious and unsolvable problem, I think it comes from a design flaw in the Amazon API, here is the problem:
I access the getDefinitionsProductType API to get the product type, which contains the "Variation_Theme" attribute.
There is a fatal problem with this attribute. It returns all variant themes, including variant themes that do not belong to this product type. I cannot judge what variant themes this product type can have.
The "description" of the "Variation_Theme" attribute tells me: /Specifies the variation theme that the product will use. The theme attribute must be applied to all products in the group./
I'm also not sure what the corresponding theme attributes of the variant theme are? They do not have corresponding keys.
This makes it impossible for me to upload multiple variants through the product type JsonSchema.
It requires me to fill in the variant theme-related attributes, but it doesn't pass me enough information to let me know what the variant theme and variant attributes I should fill in are, which is simply incredible.

**Comment by wszwc** [comment]
@chapmanjw @leonardonakagawa @rugved1991
Can you please give a reasonable and complete description? This makes JsonSchema completely unable to be used to create an html form that submits multiple variations. I think this question is unreasonable and one that can be easily solved. why not? Please pay attention to this issue, thank you very much

**Comment by weilinggu** [comment]
Hi @wszwc 

When selecting a variation_theme value for your product, cross reference [enumDeprecated](https://developer-docs.amazon.com/sp-api/docs/product-type-definition-meta-schema#enumdeprecated) in $lifecycle within variation_theme. [enumDeprecated](https://developer-docs.amazon.com/sp-api/docs/product-type-definition-meta-schema#enumdeprecated) indicates which enumeration values are considered deprecated and should be avoided. After crossing referencing, you should get a bit more clarity on which enum value to use for variation theme.

After selecting a variation theme for child listings item, corresponding properties will become required. For example, if variation_theme property is set to SIZE/COLOR/NUMBER_OF_ITEMS, then property number_of_items, color, shirt_size will become required (reference Product Type Definition JSON schema to find out what properties will become required based on selected variation theme).

Thanks
Weiling
Selling Partner Developer Services
